modules/email-service/impl/src/main/java/com/innowise/educationalsystem/config/KafkaConsumerConfig.java

```diff
@@ -0,0 +1,92 @@
+package com.innowise.educationalsystem.config;
+
+import com.innowise.educationalsystem.dto.AttemptRequest;
+import com.innowise.educationalsystem.dto.AttemptResponse;
+import com.innowise.educationalsystem.dto.MailKafkaHeaders;
+import com.innowise.educationalsystem.dto.MailRequest;
+import com.innowise.educationalsystem.dto.MailResponse;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.kafka.annotation.EnableKafka;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+
+@Slf4j
+@EnableKafka
+@Configuration
+@RequiredArgsConstructor
+@PropertySource(value = "classpath:kafka-props.yml", factory = YamlPropertySourceFactory.class)
+public class KafkaConsumerConfig {
+
+    @Value("${spring.kafka.bootstrap-servers}")
+    private String bootstrapAddress;
+
+    @Value("${kafka.consumer.mail.default-group-id}")
+    private String groupId;
+
+    private final KafkaTemplate<String, MailResponse> mailResponseKafkaTemplate;
+    private final KafkaTemplate<String, AttemptResponse> attemptResponseKafkaTemplate;
+
+    @Bean
+    public ConsumerFactory<String, MailRequest> mailRequestConsumerFactory() {
+        Map<String, Object> props = new HashMap<>();
```

Comment:
We can use KafkaProperties to build propertyMap for consumer | producer. I assume it's a bit pretty plus u shouldn't emphasize some basic metadata (like this: BOOTSTRAP_SERVERS_CONFIG) for each of the factories.